src/protocol/modulus_conversion/specialized_mul.rs

```diff
@@ -244,52 +247,63 @@ pub mod tests {
 
     #[tokio::test]
     async fn specialized_1_parallel() -> Result<(), BoxError> {
+        const ROUNDS: usize = 10;
         let world: TestWorld = make_world(QueryId);
         let context = make_contexts::<Fp31>(&world);
         let mut rng = rand::thread_rng();
 
-        let mut inputs = Vec::with_capacity(10);
-        let mut futures = Vec::with_capacity(10);
+        let mut inputs = Vec::with_capacity(ROUNDS);
+        let mut a_shares = Vec::with_capacity(ROUNDS);
+        let mut b_shares = Vec::with_capacity(ROUNDS);
+        let mut futures = Vec::with_capacity(ROUNDS);
 
-        for i in 0..10_u32 {
+        for _ in 0..ROUNDS {
             let a = rng.gen::<Fp31>();
             let b = rng.gen::<Fp31>();
 
             inputs.push((a, b));
 
-            let record_id = RecordId::from(0_u32);
-
-            let iteration = format!("{}", i);
-
-            futures.push(try_join_all(vec![
+            a_shares.push([
+                Replicated::new(a, Fp31::ZERO),
+                Replicated::new(Fp31::ZERO, Fp31::ZERO),
+                Replicated::new(Fp31::ZERO, a),
+            ]);
+            b_shares.push([
+                Replicated::new(Fp31::ZERO, b),
+                Replicated::new(b, Fp31::ZERO),
+                Replicated::new(Fp31::ZERO, Fp31::ZERO),
+            ]);
+        }
+        for i in 0..ROUNDS {
```

Comment:
One of the ugly test changes: when passing by reference, you need to hold the original values down somehow.  Which leads to this.